### ISSUE TITLE
Nearest text element announced when form field has placeholder text and no label in IE 11

### ISSUE THREAD

**Issue by tactics2** [FreedomScientific/VFO-standards-support#113]
### Summary

When a text input has placeholder text and no associated label, JAWS announces the nearest text element (td contents, paragraph, span, etc.) on form field focus. Example:

```html
<p>Text input with placeholder only</p>
<input type="text" placeholder="First Name">
```

is announced as "Text input with placeholder only edit, type in text"

### Expected result

Nearby text elements should not be announced on form field focus. In the absence of a label or aria-label, placeholder text should be announced. In the above example, JAWS should announce the field as "First Name edit, type in text"

### Example

https://s.codepen.io/tactics/debug/XPVqpe/DqkDdKYvGwJk

#### JAWS version and build number

2018.1808.10
 
#### Operating System and version

Windows 10 Home 64 bit

#### Browser and version:

IE 11 version 11.228.17134.0

**Comment by JasonAment** [comment]
@tactics2 - It appears that the codepen has expired or been deleted. Do you have a code sample that I could use?

**Comment by JAWS-test** [comment]
In my opinion, this is not a bug in JAWS. According to https://www.w3.org/TR/accname-1.1/, placeholder should not be used as a label. So the input field has no label. In this case, JAWS will output the surrounding text to allow blind people to perceive the context of the field.

Of course, it would be nice if JAWS also outputs the value in the placeholder in the case of a placeholder and missing labeling. This would also make the output more consistent (with the other output of placeholders and the output in other browsers). But in itself it is a problem of non-valid code if a form field has no label.

**Comment by jducrot** [comment]
@BenKeyFSI Can you please take a look at this ticket and let me know if it's a valid JAWS bug or if it can be closed? Thanks

**Comment by JAWS-test** [comment]
@jducrot 
Even if placeholder is not mentioned in Accname, according to https://www.w3.org/TR/html-aam-1.0/#input-type-text-input-type-password-input-type-search-input-type-tel-input-type-url-and-textarea-element the placeholder attribute should be used to determine the label of input fields. Unfortunately I was not aware of this when I made my comment a year ago

**Comment by carmacleod** [comment]
@jducrot
As @JAWS-test mentioned, the HTML-AAM spec allows placeholder to be used as a text/textarea input's label as a last resort, after title. So you may want to reopen this issue (or create a new one) to re-assess JAWS' handling of the placeholder attribute on all platforms (not just IE).

Here's the relevant content from the HTML-AAM spec:
5.1.1 input type="text", input type="password", input type="search", input type="tel", input type="email", input type="url" and textarea Element Accessible Name Computation
1. If the control has an aria-label or an aria-labelledby attribute the accessible name is to be calculated using the algorithm defined in Accessible Name and Description: Computation and API Mappings 1.1.
2. Otherwise use the associated label element(s) accessible name(s) - if more than one label is associated; concatenate by DOM order, delimited by spaces.
3. If the accessible name is still empty, then: use the control's title attribute.
4. Otherwise use the control's placeholder attribute.
5. If none of the above yield a usable text string there is no accessible name.

In case it's helpful, here are some TPG test pages that contain placeholder test cases:
- [placeholder attribute tests (only using input type="text")](https://thepaciellogroup.github.io/AT-browser-tests/test-files/placeholder-att.html)
- [input type="text"](https://thepaciellogroup.github.io/AT-browser-tests/test-files/input-text.html)
- [input type="password"](https://thepaciellogroup.github.io/AT-browser-tests/test-files/input-password.html)
- [textarea](https://thepaciellogroup.github.io/AT-browser-tests/test-files/textarea.html)

**Comment by jducrot** [comment]
@carmacleod thank you for the heads up. I am re-opening the ticket and will re-evaluate it. I think the confusion might have stemmed from the lack of a code example. I will recreate a code pen and document updated behavior in the latest version of JAWS across major browsers.

**Comment by carmacleod** [comment]
Thanks, @jducrot !
It may be worth noting that [CORE-AAM maps aria-placeholder](https://www.w3.org/TR/core-aam/#ariaPlaceholder) (and therefore HTML placeholder) to the following:

MSAA + IAccessible2 | UIA | ATK/AT-SPI | AX API
-- | -- | -- | --
Object Attribute:  placeholder-text:value | Property: HelpText: value  (before 2018, was Property: AriaProperties.placeholder: value) | Object Attribute:  placeholder-text:value | Property:  AXPlaceholderValue: value

so if placeholder isn't used as the label (which should almost never happen except as a last resort), then JAWS can look at those mappings, and consider announcing the placeholder text as "Hint text" or similar.

**Comment by jducrot** [comment]
I've created a codepen which I believe recreates the situation as described above: https://cdpn.io/TPG/debug/004a3800229efc67d0235627adf7eba7

In this version, the first input is a properly labeled "First Name" text input to act as the baseline. The second text input has dummy text in front of it acting as a visual label and it's not semantically tied to the nearby input in any way. That input does have a placeholder value of "last name".

I proceeded to test this setup using all possible combinations of the screen readers and browsers on a Windows 10 machine below:

- JAWS 2021.2011.16
- NVDA 2020.3
- Chrome 87.0.4280.66
- Edge 87.0.664.55
- FireFox 83.0
- IE 11.1198.18362

Out of all these pairings, I can can confirm that the bug as described still occurs with the latest version of JAWS & IE 11. Using this setup, the accName of the input is announced using the nearby text instead of the placeholder (the placeholder value is not announced as all). This is the only combination or screen reader & browser where I noticed this issue.

NOTE: Although the JAWS version I used is different from the one reported in the original ticket, we're required to use the latest version of JAWS at the time of testing to validate bug reports.

**Comment by TPGJRogers** [comment]
Issue is marked IE11. Closing.
